src/rapids_metadata/http.py

```diff
@@ -0,0 +1,31 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import urllib.request
+
+import pydantic
+
+from .metadata import RAPIDSMetadata
+
+
+GITHUB_METADATA_URL = "https://raw.githubusercontent.com/rapidsai/rapids-metadata/main/rapids-metadata.json"
+
+
+def fetch_from_url(url: str) -> RAPIDSMetadata:
+    with urllib.request.urlopen(url) as f:
+        return pydantic.TypeAdapter(RAPIDSMetadata).validate_json(f.read())
+
+
+def fetch_from_github() -> RAPIDSMetadata:
+    return fetch_from_url(GITHUB_METADATA_URL)
```

Comment:
I don't think the public API of this project should support 2 different ways to get this data. Until we have a specific reason to want to pass some other URL besides the default one here, I think we should keep `fetch_from_url()` private in the API here.

Would you consider the following set of changes?

* change `fetch_from_url()` to `_fetch_from_url()`
* change `fetch_from_github()` to like `fetch_latest()` (my preference) or `fetch_remote()` or something that doesn't explicitly mention the issue
* add an `__all__` entry in this module containing only that one public function

Anything consuming this as a library shouldn't really care that it's *specifically GitHub*, but just that it's an unauthenticated network call. Changing the name to not be GitHub-specific would allow us to swap out the implementation to something like hosting the file on S3 without a breaking API change.

Reply:
Done, and also renamed the module to `rapids_metadata.remote`.

Reply:
great, thanks!